pkg/server/event_handler.go

```diff
@@ -348,19 +348,35 @@ func parseGitHubSource(ctx context.Context, data []byte, objAttrs map[string]str
 
 	var r v1alpha1.GitHubSource
 
-	if c, found := pm.Metadata[MetadataKeyGitHubCommit]; found {
+	// Set github-commit. Response with error message when github-commit can not be found.
+	c, found := pm.Metadata[MetadataKeyGitHubCommit]
+	if !found {
+		return nil, fmt.Errorf("github-commit not found")
```

Comment:
I think we don't have to return error here. The source may not be from github. Just adding logging info should be enough.

Reply:
Agreed, these fields can be empty, we just want a log when these fields are missing.

Reply:
Updated & marked as resolved.